cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/input/MultiInputFormat.java

```diff
@@ -82,13 +83,17 @@
   public RecordReader<K, V> createRecordReader(InputSplit split,
                                                TaskAttemptContext context) throws IOException, InterruptedException {
     MultiInputTaggedSplit taggedInputSplit = (MultiInputTaggedSplit) split;
-    ConfigurationUtil.setAll((taggedInputSplit).getInputConfigs(), context.getConfiguration());
+    Job jobCopy = new Job(context.getConfiguration());
+    Configuration confCopy = jobCopy.getConfiguration();
+    ConfigurationUtil.setAll((taggedInputSplit).getInputConfigs(), confCopy);
+    TaskAttemptContext taskAttemptContextCopy = new TaskAttemptContextImpl(confCopy, context.getTaskAttemptID(),
+                                                                           new WrappedStatusReporter(context));
```

Comment:
Why wrapping of the status reporter is needed? If everything is just a pass-through, you can just pass in `context`, right?

Reply:
I don't think `TaskAttempContext` is a `StatusReporter` type. https://hadoop.apache.org/docs/r2.8.0/api/index.html?org/apache/hadoop/mapreduce/TaskAttemptContext.html

Reply:
I see. Interesting that they shared the same set of methods...